addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1225,12 +1225,12 @@ def test_mail_composer_wtpl_recipients_email_fields(self):
         #   there are more partners than email to notify;
         self.assertEqual(len(self._new_mails), 2, 'Should have created 2 mail.mail')
         self.assertEqual(
-            len(self._mails), (len(new_partners) + 2) * 2,
-            f'Should have sent {(len(new_partners) + 2) * 2} emails, one / recipient ({len(new_partners)} mailed partners + partner_1 + partner_2) * 2 records')
-        for record in self.test_records:
+            len(self._mails), len(new_partners) + 2,
+            f'Should have sent {len(new_partners) + 2} emails, one / recipient ({len(new_partners)} mailed partners + partner_1 + partner_2)')
+        for record, status in zip(self.test_records, ('sent', 'cancel')):
```

Comment:
If you want to write funny tests, you could update ``test_mailing_w_blacklist_opt_out`` in test_mass_mailing, checking a bit how opt_out / blacklist detection works with multi emails / formatted emails (maybe a new test to split a bit tests and avoid having too much stuff in tests).